### PR TITLE
Update to ROM 5.0

### PR DESCRIPTION
**Pull request by v-kolesnikov**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,6 +12,10 @@ gem 'rom', git: 'https://github.com/rom-rb/rom', branch: 'master' do
   gem 'rom-mapper'
 end
 
+gem 'dry-logic',  '~> 1.0', github: 'dry-rb/dry-logic',  branch: 'master'
+gem 'dry-struct', '~> 1.0', github: 'dry-rb/dry-struct', branch: 'master'
+gem 'dry-types',  '~> 1.0', github: 'dry-rb/dry-types',  branch: 'master'
+
```
Comment (solnic) [inline Gemfile]: All three gems are now released, so this can be removed.

File: .travis.yml [hunk 2]
```diff
@@ -17,7 +17,6 @@ rvm:
   - 2.6.0
   - 2.5.3
   - 2.4.5
-  - 2.3.8
```
Comment (v-kolesnikov) [inline .travis.yml]: Is no longer supported.

File: .travis.yml [hunk 3]
```diff
@@ -17,8 +17,7 @@ rvm:
...
-  - jruby-9.1.12.0
+  - jruby-head
```
Comment (solnic) [inline .travis.yml]: Why `jruby-head`?

File: rom-elasticsearch.gemspec [hunk 4]
```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ['lib']
 
-  spec.add_runtime_dependency 'rom-core', '~> 4.1'
+  spec.add_runtime_dependency 'rom-core', '~> 5.0'
```
Comment (solnic) [inline rom-elasticsearch.gemspec]: Please change to `~> 5.0`, `>= 5.0.1` because I messed up `5.0.0` 😢

File: spec/unit/rom/elasticsearch/relation/dataset_spec.rb [hunk 5]
```diff
@@ -34,15 +34,17 @@
...
           attribute :_metadata, ROM::Types::Hash,
-                    read: ROM::Types::Hash.symbolized(_index: 'string')
+                    read: ROM::Types::Hash.schema(_index: ROM::Types::String)
+                                          .with_key_transform(&:to_sym)
         end
       end
 
       users.create_index
       users.command(:create).call(id: 1, name: 'Jane')
+      users.refresh
```
Comment (solnic) [inline spec/unit/rom/elasticsearch/relation/dataset_spec.rb]: Why was this added?

Comment (v-kolesnikov) [reply spec/unit/rom/elasticsearch/relation/dataset_spec.rb]: `create` command is asynchronous, that's why. I assume it just was missed. I ran tests before commit and came across this.

File: .travis.yml (hunk 3 above)
Comment (v-kolesnikov) [reply .travis.yml]: I think it's the most useful option to have on CI because it will update automatically.

Comment (solnic) [reply .travis.yml]: We run specs against released versions of `jruby` so it would be better to use `9.2.7.0` here (although it's kinda borked on travis now 😓 )
